src/include/common/data_chunk/data_chunk_collection.h

```diff
@@ -5,36 +5,39 @@
 namespace kuzu {
 namespace common {
 
-// TODO(Guodong/Ziyi): We should extend this to ColumnDataCollection, which takes ResultSet into
-// consideration for storage and scan.
+// TODO(Guodong): Should rework this to use ColumnChunk.
 class DataChunkCollection {
 public:
     explicit DataChunkCollection(storage::MemoryManager* mm);
 
     void append(DataChunk& chunk);
-    void append(std::unique_ptr<DataChunk> chunk);
-    std::vector<common::DataChunk*> getChunks() const;
-
+    inline const std::vector<common::DataChunk>& getChunks() const { return chunks; }
+    inline std::vector<common::DataChunk>& getChunksUnsafe() { return chunks; }
     inline uint64_t getNumChunks() const { return chunks.size(); }
-    inline DataChunk* getChunk(uint64_t idx) const {
+    inline const DataChunk& getChunk(uint64_t idx) const {
+        KU_ASSERT(idx < chunks.size());
+        return chunks[idx];
+    }
+    inline DataChunk& getChunkUnsafe(uint64_t idx) {
         KU_ASSERT(idx < chunks.size());
-        return chunks[idx].get();
+        return chunks[idx];
     }
     inline void merge(DataChunkCollection* other) {
         for (auto& chunk : other->chunks) {
-            append(std::move(chunk));
+            merge(std::move(chunk));
         }
     }
+    void merge(DataChunk chunk);
 
 private:
-    DataChunk* allocateChunk(DataChunk& chunk);
+    void allocateChunk(DataChunk& chunk);
 
     void initTypes(DataChunk& chunk);
 
 private:
     storage::MemoryManager* mm;
     std::vector<LogicalType> types;
-    std::vector<std::unique_ptr<DataChunk>> chunks;
+    std::vector<DataChunk> chunks;
```

Comment:
Windows CI is failing because MSVC is trying to generate a default copy constructor for `DataChunkCollection`. I'm not really sure why, but deleting the copy constructor fixes the issue (e.g. with `DELETE_COPY_DEFAULT_MOVE` as in `DataChunk`).

Reply:
ha. thanks! Will fix it.